soap_to_rest/wsdl_service.py

```diff
@@ -0,0 +1,63 @@
+"""Module for working with a wsdl and resulting web service"""
+import logging
+from xml.sax import SAXParseException
+
+from suds import MethodNotFound, WebFault
+from suds.client import Client
+from suds.transport import TransportError
+from suds.wsse import Security, UsernameToken
+
+
+class WsdlError(Exception):
+    """Simple wrapper for any errors occurring during web service invocation"""
+
+    def __init__(self, wsdl_error):
+        super().__init__()
+        self.wrapped = wsdl_error
+
+
+def invoke_action(url, action, params, auth=None):
+    """Invokes an action contained in the provided WSDL url using provided parameters"""
+    try:
+        if auth:
+            security = Security()
+            token = UsernameToken(auth["username"], auth["password"])
+            security.tokens.append(token)
+            client = Client(url, wsse=security)
+        else:
+            client = Client(url)
+
+        return client.service.__getattr__(action)(**params)
```

Comment:
this could probably be changed to `getattr(client.service, action)` now that we know 1% more python than before